### PR TITLE
FAI-7315 - CircleCI: fetch test results

### PR DESCRIPTION
**Pull request by tovbinm**
## Description

Fetch test results uploaded to Circle CI.
1. Added a new `tests` stream
2. Added a new converter for the `tests` stream that writes Test Execution results into Faros

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Additional context
1. https://circleci.com/docs/collect-test-data/
2. https://circleci.com/docs/api/v2/index.html#operation/getTests

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts [hunk 1]
```diff
@@ -67,8 +77,73 @@ export class Pipelines extends CircleCIConverter {
...
+      }
+    } catch (e: any) {
+      const msg = e?.message ?? JSON.stringify(e);
+      ctx.logger.debug(`Failed to process ${artifact.url}: ${msg}`);
+    }
+    return [];
+  }
+
+  convertTestResult(tr: any): ReadonlyArray<DestinationRecord> {
+    // TODO: implement
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts:146]: @willmarks here is where I would like to get your help. Similar to what we did here (https://github.com/faros-ai/faros-test-results-reporter/blob/main/src/test-results-reporter.ts#L98) with TestExecution event, but now I need to create the models directly.

**Comment by tovbinm** [comment]
Ready for review!

File: sources/circleci-source/src/circleci/circleci.ts [hunk 2]
```diff
@@ -214,9 +225,26 @@ export class CircleCI {
       (item: any) => item
     );
   }
+
+  @Memoize(
+    (projectSlug: string, jobNumber: string) => `${projectSlug}_${jobNumber}`
+  )
```
Comment (willmarks) [inline sources/circleci-source/src/circleci/circleci.ts:231]: Do we need to memoize this?

**Review by willmarks — APPROVED** [review]

File: sources/circleci-source/src/circleci/circleci.ts (hunk 2 above)
Comment (tovbinm) [reply sources/circleci-source/src/circleci/circleci.ts:231]: Probably not. Feel free to remove and push to this branch.

Comment (tovbinm) [reply sources/circleci-source/src/circleci/circleci.ts:231]: Removed

**Pull request merged**